### ISSUE TITLE
missing information in MLTraining outputs paths

### ISSUE THREAD

**Issue by mafrahm** [columnflow/columnflow#234]
At the moment, only the version parameter is encoded in the `MLTraining` output path.

We should add the used ml_model and configs to the `store_parts` and probably even the used calibrators/selectors/producers

**Comment by dsavoiu** [comment]
This is partially addressed in PR #237, which adds a config identifier and the `ml_model` to the output path. Adding the the calibrators/selectors/producers would require some refactoring, one reason being that `MLTraining` accepts multiple `configs` and so cannot inherit from the calibrator/selector/producer mixins, as these explicitly require a single `config`.

Also to note is that in PR #237 the `configs` string representation directly joins together the used-supplied strings. To prevent overly long output paths, and for consistency with other places where potentially long lists of strings are encoded in the output, this could be changed to use only the first config string, followed by the total number of configs and a unique hash.

**Comment by mafrahm** [comment]
I just checked the different implementations of `store_parts`. For datasets and processes, we use length + hash as a representation, e.g.
```python
processes_repr = f"{len(self.processes)}_{law.util.create_hash(self.processes)}"
```

But for calibrators and producers, we use the first 5 instances + hash, e.g.
```python
part = "__".join(self.producers[:5])
if len(self.producers) > 5:
    part += f"__{law.util.create_hash(self.producers[5:])}"
```
Where the final string we add to the store parts is `f"prod__{part}"`

For the output parts in the MLTraining this gets a bit more complicated because you can request different calibs/selectors/producers per config (but per default it's the same one for each config).

(For simplicity, I'll only talk about producers in the following)
One way of proceeding could be:
- first checking if the producers are the same for each config
- if that's true: use the representation as we did for other producer tasks (but maybe only the first 1-2 instances before adding a hash)
- if that's false: add the number of producers per config, flatten the producers and then convert to a hash

e.g. for
```python
# three configs, always the same producer
producers = (("prod",), ("prod",), ("prod",))
```
the representation simply becomes `"prod"`, while something like
```python
# three configs with different producers
producers = (("prod_UL16",), ("prod_UL17",), ("prod_UL18",))
```
would become
```python
f"111_{law.util.create_hash(("prod_UL16", "prod_UL17", "prod_UL18"))}"
```
